code/crates/engine/src/consensus.rs

```diff
@@ -663,6 +675,7 @@ where
             .await
         {
             error!(%height, "Error when processing received synced block: {e}");
+            return Err(e.into());
```

Comment:
Just a suggestion: replace the content of this function with the following (Github UI does not allow me to select the lines to do the suggestion). It will skip one level of error message, but I think that's fine.
```
        self.process_input(
            myself,
            state,
            ConsensusInput::SyncValueResponse(CoreValueResponse::new(
                peer,
                value.value_bytes.clone(),
                value.certificate.clone(),
            )),
        )
        .await
        .map_err(|e| e.into())
```